scripts/install_appenginesdk.py

```diff
@@ -0,0 +1,1400 @@
+            if len(step) == 2:
+                if isinstance(step[1], BuildstepStep):
+                    name, step = step
+                    step.name = name
+                elif isinstance(step[1], _CALLABLE_TYPES):
+                    name, func = step
+                    step = BuildstepStep(
+                        name=name, func=func, args=args, kwargs=kwargs)
+                else:
+                    raise ValueError(step)
+            else:
+                raise ValueError(
+                    ('step', step, 'is not a 2-tuple/list'))
+        elif isinstance(step, BuildstepResult):
+            result = step
+            # XXX
+            step = BuildstepStep(
+                name=result.name,
+                func=lambda x: result,
+                result=result)
+        elif isinstance(step, _CALLABLE_TYPES):
+            func = step
+            step = BuildstepStep(
+                name=get_obj_name(func),
+                func=func)
+
+        else:
+            raise ValueError(
+                ('step', step, 'is not a BuildstepStep or a 2-tuple/list'))
+        log.info(('step', (step)))
+        if step.name is None:
+            step.name = get_obj_name(func)
+        return super(BuildstepPlan, self).append(step)
+
+    def extend(self, results):
+        for result in results:
+            self.append(result)
+
+    @property
+    def data(self):
+        return self
+
+    def __print_all(self, argv,
+                    printheader=True,
+                    prefix=None,
+                    hdrstr=None,
+                    objs=None,
+                    depth=0):
+        indentstr = '  ' * depth
+        hdrstr = prefix if hdrstr is None else hdrstr
+        if printheader:
+            _hdrstr = "##### {!s}".format(
+                (hdrstr + ' ') if hdrstr is not None else '')
+            yield indentstr + ""
+            yield indentstr + _hdrstr
+            yield indentstr + '#' * len(_hdrstr)
+        else:
+            yield ''
+        yield indentstr + '- ### args: {!r}'.format(argv)
+        if objs is None:
+            objs = self
+        for obj in objs:
+            try:
+                # if isinstance(obj, BuildstepStep):
+                if hasattr(obj, 'result'):
+                    result = obj.result
+                # if isinstance(obj, BuildstepResult):
+                elif hasattr(obj, 'data'):
+                    result = obj
+                elif isinstance(obj, (tuple, list)) and len(obj) == 2:
+                    result = obj[1]
+                else:
+                    raise ValueError((type(obj), obj))
+                # TODO step._repr_checkbox_
+                if result is None:
+                    if obj:
+                        if hasattr(obj, '_repr_checkbox_'):
+                            yield indentstr + obj._repr_checkbox_()
+                        else:
+                            yield indentstr + '+ [ ] {!s}'.format(obj)
```

Comment:
- [ ] UBY: trace ``name`` through
  - a BuildstepResult may not have an explicit name defined (e.g. defaults to ``func.func_name``)
    - [x] explicit name='check'
    - [x] explicit name='install'